src/compiler/scala/tools/nsc/transform/async/TransformUtils.scala

```diff
@@ -133,13 +133,14 @@ private[async] trait TransformUtils extends AsyncTransformStates {
 
     override def traverse(tree: Tree): Unit = {
       tree match {
-        case cd: ClassDef          =>
+        case cd: ClassDef                                                              =>
           if (cd.symbol.isAnonymousClass) function(cd)
           else if (cd.symbol.isModuleClass) nestedModuleClass(cd)
           else nestedClass(cd)
-        case dd: DefDef            => nestedMethod(dd)
-        case fun: Function         => function(fun)
-        case _                     => super.traverse(tree)
+        case dd: DefDef                                                                => nestedMethod(dd)
+        case fun: Function                                                             => function(fun)
+        case Apply(qual, arg :: Nil) if qual.symbol == definitions.Object_synchronized => synchronizedCall(arg)
```

Comment:
```suggestion
        case Apply(TypeApply(fun, _), arg :: Nil) if fun.symbol == definitions.Object_synchronized => synchronizedCall(arg)
```

Your version also works as the `TypeApply`'s symbol is also `Object_synchronized`, but should we rely on that?